### PR TITLE
Accelerate SequencePool Op on SUM mode of CPU

### PR DESCRIPTION
**Pull request by velconia**
test=develop

### CODE REVIEW

File: paddle/fluid/operators/math/sequence_pooling.cc [hunk 1]
```diff
@@ -191,7 +194,14 @@ class SequencePoolFunctor<platform::CPUDeviceContext, T> {
...
         out_e.device(place) = in_e.mean(Eigen::array<int, 1>({{0}}));
       } else if (pooltype == "SUM") {
-        out_e.device(place) = in_e.sum(Eigen::array<int, 1>({{0}}));
+        if (h > 0) {
+          const T* in_data = in_t.data<T>();
+          T* out_data = out_t.mutable_data<T>(context.GetPlace());
+          blas.VCOPY(w, in_data, out_data);
+          for (int64_t r = 1; r != h; ++r) {
+            blas.AXPY(w, 1., in_data + r * w, out_data);
+          }
```
Comment (sneaxiy) [inline paddle/fluid/operators/math/sequence_pooling.cc:203]: Interesting codes. It is not intuitive to know whether it is faster than the previous implementation. Is there any verifying data?

File: paddle/fluid/operators/math/sequence_pooling.cc [hunk 2]
```diff
@@ -191,7 +194,14 @@ class SequencePoolFunctor<platform::CPUDeviceContext, T> {
...
+        }
       } else if (pooltype == "SQRT") {
         out_e.device(place) = in_e.sum(Eigen::array<int, 1>({{0}})) /
```
Comment (sneaxiy) [inline paddle/fluid/operators/math/sequence_pooling.cc:206]: How about change "SQRT" together? It is almost the same.

File: paddle/fluid/operators/math/sequence_pooling.cc (hunk 1 above)
Comment (velconia) [reply paddle/fluid/operators/math/sequence_pooling.cc:203]: the speed of ```blas``` will benefit from the ```SIMD``` and ```AVX``` instrincs

File: paddle/fluid/operators/math/sequence_pooling.cc (hunk 2 above)
Comment (velconia) [reply paddle/fluid/operators/math/sequence_pooling.cc:206]: I didn't test ```SQRT``` actually and maybe it's better to test in real models and change the code

**Review by sneaxiy — APPROVED** [review]
Excellent

**Pull request merged**